server/expression.js

```diff
@@ -167,6 +182,51 @@ expression.getTfaProfile_ = function(fileName, geneNames, conditionNames) {
 };
 
 /**
+ * Reads the TFA matrix in text file.
+ * @param {string} fileName Path to the TFA matrix file.
+ * @return {!expression.Matrix}
+ * @private
+ */
+expression.readTfaMatrix_ = function(fileName) {
```

Comment:
The readTfaMatrix_ function returns the whole matrix, and readExpression_ returns a selected sub matrix.